overlord/ifacestate/handlers.go

```diff
@@ -303,3 +306,24 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 	setConns(st, conns)
 	return nil
 }
+
+func (m *InterfaceManager) doDiscardNamespace(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapSetup, err := snapstate.TaskSnapSetup(task)
+	if err != nil {
+		return err
+	}
+
+	snapName := snapSetup.Name()
+
+	// Run "snap-discard-ns $SNAP_NAME" and ignore the error code.  This
+	// command is not meant to always succeed because the namespace may not
+	// have been created yet in practice (e.g. the snap was removed before it
```

Comment:
That sounds bad. If we want to ignore a particular condition, we should teach snap-discard-ns to ignore that one particular condition, probably behind a flag. Or even better: why do we run this command if the namespace is not there? Simply ignoring the fact it fails in whichever way will hide real bugs. The tool might not even be there, and this will still pretend it's all good.

Let's please get both the error code and the combined output, and actually fail the task if it doesn't work. We want to know.